### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by PepijnSenders**
This is the first commit for the open-source webpack plugin to list a project's dependencies in the `dist` folder.

We will use this for our fragments to propagate which dependencies they need to work.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<p align="center">
+    <a href="https://hellofresh.com">
+        <img width="120" src="https://www.hellofresh.de/images/hellofresh/press/HelloFresh_Logo.png">
+    </a>
+</p>
+
+# Dependency Manifest
```
Comment (nik-kor) [inline README.md]: what is the use case of this plugin? 
I mean, it's good to detail what is it all about.

P.S. I haven't read the code - just tests - can't understand why not use package.json)

Comment (PepijnSenders) [inline README.md]: I need to write a more proper README tomorrow, but the main goal is that you can get the dependencies from one entry point from webpack, which is not the same as what is listed inside the package.json.

File: ci/scripts/install.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#! /bin/sh
```
Comment (nik-kor) [inline ci/scripts/install.sh]: Why don't just use `scripts` section in `package.json`?

it is a more standard way for it and these scripts are actually one-liners

Comment (nik-kor) [inline ci/scripts/install.sh]: ah.... maybe it's hard configure ci then

Comment (PepijnSenders) [inline ci/scripts/install.sh]: It's nice to have it this way, to have more flexibility. I still need to copy the folder after :), also I don't think the `npm scripts` are 1 to 1 connected to the CI.

File: ci/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
+---
```
Comment (nik-kor) [inline ci/pipeline.yml]: if it's considered as an open-source project maybe it's better to use some free tools for it like Travis?
Actually, I'm not familiar with concourse.

E.g. if somebody will fork the project can he use CI tools then?

Comment (PepijnSenders) [inline ci/pipeline.yml]: Yeah, that might be a good idea, the project inside concourse won't be Open Source. Do you maybe have an example configuration lying around from a node project in travis? With publishing the repo to NPM.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "dependency-manifest-webpack-plugin",
+  "version": "0.0.1",
+  "description": "Dependency Manifest",
+  "main": "index.js",
+  "scripts": {
+    "test": "mocha -t 10000 --require babel-core/register"
```
Comment (PepijnSenders) [inline package.json]: Good spot, I had this for the webpack generation time, can probably turn this down to 5s.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "lodash": "~4.17.0"
+  },
+  "devDependencies": {
+    "babel-preset-stage-0": "~6.16.0",
+    "chai": "~3.5.0",
+    "memory-fs": "~0.3.0",
+    "mocha": "~3.1.2",
+    "mock-fs": "~3.12.1",
+    "muject": "~1.2.0",
+    "webpack": "~1.13.3"
```
Comment (nik-kor) [inline package.json]: haha. we are using webpack 2 already

File: ci/scripts/install.sh (hunk 2 above)
Comment (nik-kor) [inline ci/scripts/install.sh]: what kind of flexibility do you mean?) What for?)

Btw it's good to have all the scripts in one place like package.json - so you see everything - it's compact

Comment (nik-kor) [inline ci/scripts/install.sh]: I definitely agree about bash scripts if there are a lot in them - so you can't put it inside json

Comment (PepijnSenders) [inline ci/scripts/install.sh]: Yeah I know what you mean, but there's a difference between these tasks and the npm scripts, I like to have simple tasks there and combine them in these bash files :)

File: src/DependencyManifestPlugin.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+        const manifest = {};
+        listedDependencies.forEach((packageName) => {
+            manifest[packageName] = packageDependencies[packageName];
+        });
+
+        return manifest;
+    }
+
+    getOutputFile(compiler) {
```
Comment (nik-kor) [inline src/DependencyManifestPlugin.js]: test is missed for it

**Comment by nik-kor** [comment]
Great!

**Comment by nik-kor** [comment]
+1

**Comment by renannfortes** [comment]
+1

**Pull request merged**